src/btree/bt_split.c

```diff
@@ -1509,30 +1517,48 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
  *     Discard moved update lists from the original page.
  */
 static void
-__split_multi_inmem_final(WT_PAGE *orig, WT_MULTI *multi)
+__split_multi_inmem_final(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi)
 {
     WT_SAVE_UPD *supd;
+    WT_UPDATE *upd;
     uint32_t i, slot;
 
+    /* If we have saved updates, we must have decided to restore them to the new page. */
+    WT_ASSERT(session, multi->supd_entries == 0 || multi->supd_restore);
+
     /*
      * We successfully created new in-memory pages. For error-handling reasons, we've left the
      * update chains referenced by both the original and new pages. We're ready to discard the
      * original page, terminate the original page's reference to any update list we moved.
      */
-    for (i = 0, supd = multi->supd; i < multi->supd_entries; ++i, ++supd)
-        switch (orig->type) {
-        case WT_PAGE_COL_FIX:
-        case WT_PAGE_COL_VAR:
+    for (i = 0, supd = multi->supd; i < multi->supd_entries; ++i, ++supd) {
+        /* Discard the update chain that is not restored. */
+        if (!supd->restore)
```

Comment:
To be clear, are we restoring the update chain after this comment? In which case it should be worded differently:
"Discard the update chain that is not going to be restored."

Reply:
We have restored the one needed to be restored already. So we delink them from the old page here.